src/libraries/TwabLib.sol

```diff
@@ -18,435 +17,674 @@ import { ObservationLib, MAX_CARDINALITY } from "./ObservationLib.sol";
  *         TWAB checkpoint is stored in the circular ring buffer, as either a new checkpoint or
  *         rewriting a previous checkpoint with new parameters. One checkpoint per day is stored.
  *         The TwabLib guarantees minimum 1 year of search history.
+ * @notice There are limitations to the Observation data structure used. Ensure your token is
+ *         compatible before using this library. Ensure the date ranges you're relying on are
+ *         within safe boundaries.
  */
 library TwabLib {
+  /**
+   * @dev Sets the minimum period length for Observations. When a period elapses, a new Observation
+   *      is recorded, otherwise the most recent Observation is updated.
+   */
+  uint32 constant PERIOD_LENGTH = 1 days;
```

Comment:
So this value will not be set in the constructor? I suppose 1 day works for a weekly draw?

Reply:
Potential gas savings was the reason. However, in the constructor would definitely be easier to work with.
https://docs.soliditylang.org/en/v0.8.17/contracts.html#constant-and-immutable-state-variables
>  For a constant variable, the expression assigned to it is copied to all the places where it is accessed and also re-evaluated each time. This allows for local optimizations. Immutable variables are evaluated once at construction time and their value is copied to all the places in the code where they are accessed. For these values, 32 bytes are reserved, even if they would fit in fewer bytes. Due to this, constant values can sometimes be cheaper than immutable values.

The launch frequency is still tbd. As long as it's <= the draw length and is a clean fraction of the draw length then we should have no issues with data accuracy, it's more of a question of optimizing gas savings vs accuracy for unexpected frequencies.